src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -772,6 +773,39 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         }
     }
 
+    private fun checkReprAttribute(holder: RsAnnotationHolder, attr: RsAttr) {
+        if (attr.metaItem.name != "repr") return
+
+        val reprArgs = attr.metaItem.metaItemArgs?.metaItemList.orEmpty()
+
+        reprArgs.forEach {
+            val reprName = it.name ?: return
```

Comment:
This is non-local `return` directly to the caller of `checkReprAttribute`. Is it intentional or should it be a local return?

Reply:
> or should it be a local return?

In this case, let's use a simple for loop with a label:
```kotlin
label@ for (reprArg in reprArgs) {
    val reprName = reprArg.name ?: continue
    val owner = attr.owner ?: continue

    val errorText = when (reprName) {
        "C", "transparent", "align" -> when (owner) {
            is RsStructItem, is RsEnumItem -> continue@label
            else -> "$reprName attribute should be applied to struct, enum, or union"
        }
        in REPR_PRIMITIVE_REPRESENTATIONS -> when (owner) {
            is RsEnumItem -> continue@label
            else -> "$reprName attribute should be applied to enum"
        }
        "packed" -> when (owner) {
            is RsStructItem -> continue@label
            else -> "packed attribute should be applied to struct or union"
        }
        else -> {
            RsDiagnostic.UnrecognizedReprAttribute(reprArg, reprName).addToHolder(holder)
            continue@label
        }
    }

    RsDiagnostic.ReprAttrUnsupportedItem(reprArg, errorText).addToHolder(holder)
}
```

Reply:
This is intentional, I think this provides some help for `E0587`, `E0692` and other several wrong implementations (probably?

> ```kotlin
> label@ for (reprArg in reprArgs) {
>     val reprName = reprArg.name ?: continue
>     val owner = attr.owner ?: continue
> 
>     val errorText = when (reprName) {
>         "C", "transparent", "align" -> when (owner) {
>             is RsStructItem, is RsEnumItem -> continue@label
>             else -> "$reprName attribute should be applied to struct, enum, or union"
>         }
>         in REPR_PRIMITIVE_REPRESENTATIONS -> when (owner) {
>             is RsEnumItem -> continue@label
>             else -> "$reprName attribute should be applied to enum"
>         }
>         "packed" -> when (owner) {
>             is RsStructItem -> continue@label
>             else -> "packed attribute should be applied to struct or union"
>         }
>         else -> {
>           RsDiagnostic.UnrecognizedReprAttribute(reprArg, reprName).addToHolder(holder)
>             continue@label
>         }
>     }
> 
>     RsDiagnostic.ReprAttrUnsupportedItem(reprArg, errorText).addToHolder(holder)
> }
> ```

But this may be more intuitive, I will modify.